solidity/test/NetworkSettings.js

```diff
@@ -0,0 +1,87 @@
+const { contract, accounts } = require('@openzeppelin/test-environment');
+const { BN, constants, expectRevert } = require('@openzeppelin/test-helpers');
+const { expect } = require('../../chai-local');
+
+const NetworkSettings = contract.fromArtifact('NetworkSettings');
+
+const PPM_RESOLUTION = new BN(1000000);
+
+describe('NetworkSettings', () => {
+    let networkSettings;
+
+    const owner = accounts[1];
+    const nonOwner = accounts[2];
+
+    const validFeeWallet1 = accounts[3];
+    const validFeeWallet2 = accounts[4];
+
+    const validFeePortion1 = new BN(111);
+    const validFeePortion2 = new BN(222);
+
+    const invalidFeeWallet = constants.ZERO_ADDRESS;
+    const tooSmallFeePortion = new BN(0);
+    const tooLargeFeePortion = PPM_RESOLUTION.addn(1);
+
+    const expectReturn = async (method, object) => {
+        expect(JSON.stringify(await method)).to.be.equal(JSON.stringify(object));
+    };
+
+    it('should revert when creating a contract with an invalid fee wallet', async () => {
+        await expectRevert(NetworkSettings.new(invalidFeeWallet, validFeePortion1), 'ERR_INVALID_ADDRESS');
+    });
+
+    it('should revert when creating a contract with a too-small fee portion', async () => {
+        await expectRevert(NetworkSettings.new(validFeeWallet1, tooSmallFeePortion), 'ERR_INVALID_PORTION');
+    });
+
+    it('should revert when creating a contract with a too-large fee portion', async () => {
+        await expectRevert(NetworkSettings.new(validFeeWallet1, tooLargeFeePortion), 'ERR_INVALID_PORTION');
+    });
+
+    it('should revert when setting an invalid fee wallet', async () => {
+        let networkSettings = await NetworkSettings.new(validFeeWallet1, validFeePortion1);
```

Comment:
`neworkSettings` in this test and the tests below can be DRYed into a `beforeEach` callback

Reply:
Not really. See the first few `it` clauses.

Reply:
I'm not referring to the "bad" cases, but only to the "good" cases below this line. This is also why I usually wrap the "good" cases in an explicit `describe`, so that you could reuse stuff.

Reply:
So split into two describes - one for testing construction with illegal input, and one for all the other tests, and then add a `beforeEach` clause in the 2nd describe?
It will overall yield more text in this specific test, and of course, will not speed up anything.

Reply:
It's not supposed to speed anything... it's removing code duplication by moving shared code into a single locaiton

Reply:
And also separating semantically different test contexts

Reply:
Done.

Reply:
💪 💪 💪